optparse-applicative.cabal

```diff
@@ -112,6 +112,7 @@ library
   ghc-options:         -Wall
 
   build-depends:       base                            == 4.*
+                     , semigroups                      >= 0.10 && < 0.19
```

Comment:
fwiw, you can guard this via `if !impl(ghc >= 8)` if you want to avoid redundant compat dependency. Otoh,  seems pointless, since there's a `transformers-compat` dep below, so that's probably not a huge concern here.